src/Service/HealthCheck/HealthCheckHelper.cs

```diff
@@ -0,0 +1,254 @@
+                GraphQL = runtimeConfig?.Runtime?.GraphQL != null && runtimeConfig.Runtime.GraphQL.Enabled,
+                Caching = runtimeConfig?.Runtime?.IsCachingEnabled ?? false,
+                Telemetry = runtimeConfig?.Runtime?.Telemetry != null,
+                Mode = runtimeConfig?.Runtime?.Host?.Mode ?? HostMode.Development,
+            };
+        }
+
+        private void UpdateHealthCheckDetails(ref ComprehensiveHealthCheckReport ComprehensiveHealthCheckReport, RuntimeConfig runtimeConfig)
+        {
+            ComprehensiveHealthCheckReport.Checks = new List<HealthCheckResultEntry>();
+            UpdateDataSourceHealthCheckResults(ref ComprehensiveHealthCheckReport, runtimeConfig);
+            UpdateEntityHealthCheckResults(ref ComprehensiveHealthCheckReport, runtimeConfig);
+        }
+
+        private void UpdateDataSourceHealthCheckResults(ref ComprehensiveHealthCheckReport ComprehensiveHealthCheckReport, RuntimeConfig runtimeConfig)
+        {
+            if (ComprehensiveHealthCheckReport.Checks != null && runtimeConfig.DataSource?.Health != null && runtimeConfig.DataSource.Health.Enabled)
+            {
+                string query = Utilities.GetDatSourceQuery(runtimeConfig.DataSource.DatabaseType);
+                (int, string?) response = ExecuteSqlDBQuery(query, runtimeConfig.DataSource?.ConnectionString);
+                bool thresholdCheck = response.Item1 >= 0 && response.Item1 < runtimeConfig?.DataSource?.Health.ThresholdMs;
+
+                // Add DataSource Health Check Results
+                ComprehensiveHealthCheckReport.Checks.Add(new HealthCheckResultEntry
+                {
+                    Name = runtimeConfig?.DataSource?.Health?.Name ?? runtimeConfig?.DataSource?.DatabaseType.ToString(),
+                    ResponseTimeData = new ResponseTimeData
+                    {
+                        DurationMs = response.Item1,
+                        ThresholdMs = runtimeConfig?.DataSource?.Health?.ThresholdMs
+                    },
+                    Exception = !thresholdCheck ? _timeExceededErrorMessage : response.Item2,
+                    Tags = [HttpUtilities.DataSource],
+                    Status = thresholdCheck ? HealthStatus.Healthy : HealthStatus.Unhealthy
+                });
+            }
+        }
+
+        private (int, string?) ExecuteSqlDBQuery(string query, string? connectionString)
+        {
+            string? errorMessage = null;
+            if (!string.IsNullOrEmpty(query) && !string.IsNullOrEmpty(connectionString))
+            {
+                Stopwatch stopwatch = new();
+                stopwatch.Start();
+                errorMessage = _httpUtility.ExecuteDbQuery(query, connectionString);
+                stopwatch.Stop();
+                return string.IsNullOrEmpty(errorMessage) ? ((int)stopwatch.ElapsedMilliseconds, errorMessage) : (-1, errorMessage);
+            }
+
+            return (-1, errorMessage);
```

Comment:
In the case of a magic number, you are using -1 to indicate a state, you should move the value into a well-named constant.